### PR TITLE
Spectator Camera position/orientation update speedup

### PR DESCRIPTION
**Pull request by zfox23**
This new code updates the Secondary Camera's position and orientation from the C++ instead of from the JavaScript. This completely eliminates the existing problem of the camera's position being in a different location than the in-world camera model.

We achieve this by supplying an entity UUID to the secondary camera render config. The new Secondary Camera render code checks to see if such a UUID has been supplied to the config, and, if so, gets the position and orientation of that entity in the `run()` loop.

This PR also does a bunch of API cleanup and eliminates unnecessary `configure()` calls.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1203/HighFidelity-Beta-PR10856-884b8212e676076204a3262668ca22820a525879.dmg

**Comment by zfox23** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1206/HighFidelity-Beta-PR10856-884b8212e676076204a3262668ca22820a525879.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1206/HighFidelity-Beta-PR10856-884b8212e676076204a3262668ca22820a525879.exe

**Comment by zfox23** [comment]
[Investigate making camera position/orientation/fov/clip plane distances updated in C++ instead of JS (i.e. in the game loop instead of JS loop)](https://trello.com/c/VqOgswn7/220-investigate-making-camera-position-orientation-fov-clip-plane-distances-updated-in-c-instead-of-js-ie-in-the-game-loop-instead-o)

**Comment by zfox23** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1231/HighFidelity-Beta-PR10856-f765b4136dc7645238ebe8ea8b59f208effdf638.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1231/HighFidelity-Beta-PR10856-f765b4136dc7645238ebe8ea8b59f208effdf638.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1243/HighFidelity-Beta-PR10856-947a3bf8a8082ad6442ecebe668e4325a4d95bb1.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1243/HighFidelity-Beta-PR10856-947a3bf8a8082ad6442ecebe668e4325a4d95bb1.dmg

File: interface/src/SecondaryCamera.cpp [hunk 1]
```diff
@@ -83,8 +91,14 @@ class BeginSecondaryCameraFrame {  // Changes renderContext for our framebuffer
             });
 
             auto srcViewFrustum = args->getViewFrustum();
-            srcViewFrustum.setPosition(_position);
-            srcViewFrustum.setOrientation(_orientation);
+            if (_attachedEntityId != QUuid()) {
```
Comment (howard-stearns) [inline interface/src/SecondaryCamera.cpp]: !_attachedEntityId.isNull()

(I'm not certain that the compiler optimizes out the allocation in QUuid().)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1247/HighFidelity-Beta-PR10856-9bd80ec2bf938a95c7035e3912efde38b220bfdc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1255/HighFidelity-Beta-PR10856-7851c8002fae08a44cb1c1f89f8b20cb8ee3b08a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1255/HighFidelity-Beta-PR10856-7851c8002fae08a44cb1c1f89f8b20cb8ee3b08a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1287/HighFidelity-Beta-PR10856-9c96cf811e9db78935d9c0f3812cd1c0ece51138.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1287/HighFidelity-Beta-PR10856-9c96cf811e9db78935d9c0f3812cd1c0ece51138.dmg

File: interface/src/SecondaryCamera.cpp [hunk 2]
```diff
@@ -27,39 +29,32 @@ void MainRenderTask::build(JobModel& task, const render::Varying& inputs, render
...
-    textureCache->resetSpectatorCameraFramebuffer(width, height); // FIXME: Call the correct reset function based on the "destinationFramebuffer" arg
-    setEnabled(wasEnabled);
-}
-
-void SecondaryCameraRenderTaskConfig::resetSizeSpectatorCamera(int width, int height) { // Carefully adjust the framebuffer / texture.
-    resetSize(width, height);
-}
-
-class BeginSecondaryCameraFrame {  // Changes renderContext for our framebuffer and view.
+class SecondaryCameraJob {  // Changes renderContext for our framebuffer and view.
```
Comment (manapa01) [inline interface/src/SecondaryCamera.cpp]: I agree with renaming the BeginSecondaryCameraFrame to SecondaryCameraJob, it is much less confusing

File: interface/src/SecondaryCamera.cpp [hunk 3]
```diff
@@ -99,6 +100,41 @@ class BeginSecondaryCameraFrame {  // Changes renderContext for our framebuffer
...
+void SecondaryCameraJobConfig::setOrientation(glm::quat orient) {
+    if (attachedEntityId.isNull()) {
+        orientation = orient;
+        emit dirty();
+    } else {
+        qDebug() << "ERROR: Cannot set orientation of SecondaryCamera while attachedEntityId is set.";
+    }
+}
+
+void SecondaryCameraJobConfig::enableSecondaryCameraRenderConfigs(bool enabled) {
```
Comment (manapa01) [inline interface/src/SecondaryCamera.cpp]: I like this way of enabling the configs better than the previous, its much cleaner. Nice work!

File: interface/src/SecondaryCamera.cpp [hunk 4]
```diff
@@ -99,6 +100,41 @@ class BeginSecondaryCameraFrame {  // Changes renderContext for our framebuffer
     RenderArgsPointer _cachedArgsPointer;
 };
 
+void SecondaryCameraJobConfig::setPosition(glm::vec3 pos) {
+    if (attachedEntityId.isNull()) {
+        position = pos;
+        emit dirty();
```
Comment (manapa01) [inline interface/src/SecondaryCamera.cpp]: Why do we need to emit dirty()?

Comment (zfox23) [inline interface/src/SecondaryCamera.cpp]: So that `configure()` gets called again. :)

File: interface/src/SecondaryCamera.cpp (hunk 2 above)
Comment (zfox23) [inline interface/src/SecondaryCamera.cpp]: :+1:

File: interface/src/SecondaryCamera.cpp (hunk 3 above)
Comment (zfox23) [inline interface/src/SecondaryCamera.cpp]: Thanks! :)

**Pull request merged**